src/main/java/nextstep/subway/favorite/domain/Favorite.java

```diff
@@ -0,0 +1,71 @@
+package nextstep.subway.favorite.domain;
+
+import nextstep.subway.BaseEntity;
+import nextstep.subway.exception.NotFoundDataException;
+import nextstep.subway.member.domain.Member;
+import nextstep.subway.station.domain.Station;
+
+import javax.persistence.*;
+import java.util.Objects;
+
+import static nextstep.subway.exception.type.NotFoundDataExceptionType.NOT_FOUND_STATION;
+
+@Entity
+public class Favorite extends BaseEntity {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "source_id")
+    private Station source;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "target_id")
+    private Station target;
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "member_id")
+    private Member member;
+
+    protected Favorite() {}
+
+    private Favorite(Station source, Station target, Member member) {
+        this.source = source;
+        this.target = target;
+        this.member = member;
+    }
+
+    public static Favorite of(Station sourceStation, Station targetStation, Member member) {
+        validationCheckIsExistStation(sourceStation, targetStation);
+        
+        return new Favorite(sourceStation, targetStation, member);
+    }
+
+    private static void validationCheckIsExistStation(Station sourceStation, Station targetStation) {
+        if (Objects.isNull(sourceStation)) {
```

Comment:
```suggestion
        if (Objects.isNull(sourceStation) || Objects.isNull(targetStation)) {
```
이렇게 한번에 작성할 수 있지 않을까요...?
추가로, sourceStation과 targetStation이 같지 않다.. 도 가능할 것 같습니다..!